modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js

```diff
@@ -158,37 +128,46 @@ export default function CartQuickAdd() {
 	const handleAddToCartClick = () => {
 		const readySKUs = selectedSKUs.map((selectedSKU) => {
 			const {
+				discontinued,
 				id,
 				productConfiguration,
-				replacementSkuId,
+				replacementSku,
 				sku,
 				urls,
 			} = selectedSKU;
 
-			if (replacementSkuId) {
-				const replacementSKU = availableSKUs.find(
-					(availableSKU) => availableSKU.id === replacementSkuId
-				);
+			if (discontinued) {
+				const {
+					productConfiguration: replacementConfiguration,
+					sku,
+					skuId,
+					skuOptions,
+				} = replacementSku;
 
 				return {
-					...replacementSKU,
+					...selectedSKU,
 					productURLs: urls,
 					quantity: getCorrectedQuantity(
-						replacementSKU,
-						replacementSKU.sku,
+						replacementConfiguration,
+						sku,
 						cartItems
 					),
-					replacementSkuId: id,
-					settings: productConfiguration,
-					skuId: replacementSKU.id,
-					urls: replacementSKU.urls,
+					replacedSkuId: id,
+					settings: replacementConfiguration,
+					skuId,
+					skuOptions,
+					urls,
```

Comment:
```
return {
					...replacementSKU,
					quantity: getCorrectedQuantity(
						replacementConfiguration,
						sku,
						cartItems
					),
					replacedSkuId: id,
					settings: replacementConfiguration,
```

Reply:
@alessiorendina, there are side effects to calling `addToCart` method, so we need, from `replacementSKU` object, also the `price` and the `urls` properties. In the last commit, I hardcoded it to don't push breakable code.